clib/mininet_test_base.py

```diff
@@ -241,6 +246,26 @@ def _enable_event_log(self, timeout=None):
         controller.cmd(mininet_test_util.timeout_cmd(
             'nc -U %s > %s &' % (sock, self.event_log), timeout))
 
+    def _wait_until_matching_event(self, match_func, timeout=30):
+        """Return the next matching event from the event sock, else fail"""
+        assert timeout >= 1
+        assert self.event_log and os.path.exists(self.event_log)
+        for _ in range(timeout):
+            with open(self.event_log) as events:
+                for event_str in events:
+                    event = json.loads(event_str)
+                    event_id = event['event_id']
+                    if event_id <= self.prev_event_id:
+                        continue
+                    self.prev_event_id = event_id
+                    try:
+                        if match_func(event):
+                            return event
+                    except Exception:
```

Comment:
Please replace with the actual expected exception.

Reply:
Updated.